app/helpers/application_helper.rb

```diff
@@ -16,7 +16,7 @@ def round_meters_up_to_5_mi(meters)
 
   def link_to_locale(locale, label, additional_attributes = {})
     link_to(label,
-            "?" + request.query_parameters.merge("new_locale" => locale).to_query,
+            { :locale=>locale, :params=>request.query_parameters },
```

Comment:
sets the locale as part of the path instead of as a separate query param, retaining other query params